### PR TITLE
Run3-alca215 Provide possibility of using ECAL RecHit thresholds used by particle flow group in Hcal Calibration code

### PR DESCRIPTION
**Pull request by bsunanda**
#### PR description:

Provide possibility of using ECAL RecHit thresholds used by particle flow group in Hcal Calibration code. Also introduce a new modifier for ECAL_2018

#### PR validation:

Use the runTheMatrix test workflows and workflows in the test areas of Calibration/HcalAlCaRecoProducers and Calibration/HcalCalibAlgos

#### if this PR is a backport please specify the original PR and why you need to backport that PR:

Nothing special

### CODE REVIEW

**Comment by bsunanda** [comment]
@cmsbuild Please test

File: Calibration/HcalCalibAlgos/test/python/isoTrackRecoAnalysis_cfg.py [hunk 1]
```diff
@@ -26,7 +26,7 @@
 process.towerMakerAll.ecalInputs = cms.VInputTag(cms.InputTag("ecalRecHit","EcalRecHitsEB"), cms.InputTag("ecalRecHit","EcalRecHitsEE"))
 process.towerMakerAll.AllowMissingInputs = True
 
-process.load('Calibration.HcalCalibAlgos.HcalIsoTrkAnalyzer_cff')
+process.load('Calibration.HcalCalibAlgos.hcalIsoTrkAnalyzer_cff')
```
Comment (francescobrivio) [inline Calibration/HcalCalibAlgos/test/python/isoTrackRecoAnalysis_cfg.py:29]: @bsunanda sorry for the maybe trivial question, but why did you change `HcalIsoTrkAnalyzer_cff` to `hcalIsoTrkAnalyzer_cff` (and same in the other .py files in this PR)? 
In the repo i still see the file with the capital letter:
https://github.com/cms-sw/cmssw/blob/master/Calibration/HcalCalibAlgos/python/HcalIsoTrkAnalyzer_cff.py

Comment (francescobrivio) [reply Calibration/HcalCalibAlgos/test/python/isoTrackRecoAnalysis_cfg.py:29]: Sorry i just realized that it is renamed in this PR! In that case that's fine!

**Comment by francescobrivio** [comment]
+alca

 - Tests pass
 - Introduced new `ECAL_2018` modifier as followup to #36641
 - Added possibility to set ECAL RecHit thresholds directly from EventSetup, `true` by default as discussed in [slide 11](https://indico.cern.ch/event/1111295/contributions/4677212/attachments/2373966/4054986/IsoTrigger13.pdf#page=11&zoom=auto,-300,768) of Sunanda's contribution at AlCaDB meeting

**Comment by perrotta** [comment]
+operations

**Comment by perrotta** [comment]
+1
- Newly defined ECAL_2018 modifier replace where needed to HCAL_2018 (previously used also as a kind of placeholder)
